### PR TITLE
Refactor xml writer

### PR DESCRIPTION
**Pull request by jaschultz**

### CODE REVIEW

File: python_testspace_xml/testspace_xml.py [hunk 1]
```diff
@@ -33,107 +33,57 @@ def __init__(self, name, comment):
...
-        anno.setAttribute("name", self.name)
+    def write_xml(self, parent_element, dom):
+        annotation = dom.createElement("annotation")
+        annotation.setAttribute("description", self.description)
+        annotation.setAttribute("level", self.level)
+        annotation.setAttribute("name", self.name)
+
+        if self.file_path is not None:
+            if not os.path.isfile(self.file_path):
+                # write as a warning text annotation
```
Comment (ivailop) [inline python_testspace_xml/testspace_xml.py]: correct or remove this comment

File: python_testspace_xml/testspace_xml.py [hunk 2]
```diff
@@ -145,34 +95,33 @@ def __init__(self, name, status='passed'):
...
         self.duration = duration
 
-    def fail(self, reason):
+    def set_status(self, status):
+        self.status = status
+
+    def set_start_time(self, gmt_string):
+        self.start_time = gmt_string
+
+    def add_fail_annotation(self, message):
```
Comment (ivailop) [inline python_testspace_xml/testspace_xml.py]: I think the old `fail` method is better as its primary purpose is to fail the case and optionally annotate

File: python_testspace_xml/testspace_xml.py [hunk 3]
```diff
@@ -33,107 +33,57 @@ def __init__(self, name, comment):
...
+                annotation.setAttribute("file", "file://" + self.file_path)
+                annotation.setAttribute("mime_type", self.mimeType)
+                with io.open(self.file_path, 'rb') as inFile:
+                    out = BytesIO()
+                    with gzip.GzipFile(fileobj=out, mode="wb") as f:
+                        f.writelines(inFile)
+                    f.close()
+                    gzip_data = out.getvalue()
+                    b64_data = base64.standard_b64encode(gzip_data)
+                    cdata = dom.createCDATASection(b64_data.decode())
```
Comment (ivailop) [inline python_testspace_xml/testspace_xml.py]: why calling `decode()`?

Comment (jaschultz) [reply python_testspace_xml/testspace_xml.py]: The [decode ](https://docs.python.org/3.3/library/stdtypes.html#bytes.decode)is to convert b64_data from bytes to string.

**Pull request merged**